Entity/Repository/JobRepository.php

```diff
@@ -152,22 +152,9 @@ public function getOrCreateIfNotExists($command, array $args = array(), $queueNa
         return $firstJob;
     }
 
-    public function findStartableJob($queueName, array &$excludedIds = array())
+    public function findStartableJob($queueName)
     {
-        while (null !== $job = $this->findPendingJob($queueName, $excludedIds)) {
-            if ($job->isStartable()) {
-                return $job;
-            }
-
-            $excludedIds[] = $job->getId();
-
-            // We do not want to have non-startable jobs floating around in
-            // cache as they might be changed by another process. So, better
-            // re-fetch them when they are not excluded anymore.
-            $this->_em->detach($job);
-        }
-
-        return null;
+        return $this->findPendingJob($queueName);
```

Comment:
You are right about the initial design. However, the truth is isStartable() is not used by us. What the function does is simply checking all the dependent jobs is finished. And we are not using the jmb_dependencies table at all. 
https://github.com/revinate/JMSJobQueueBundle/blob/master/Entity/Job.php#L260-L269